### PR TITLE
Bugfix: conditional rendering

### PR DESCRIPTION
**Pull request by drwpow**
Fixes a bug with conditional rendering.

### CODE REVIEW

File: internal/printer/print-to-js.go [hunk 1]
```diff
@@ -303,12 +303,11 @@ func render1(p *printer, n *Node, opts RenderOptions) {
...
 
 		for c := n.FirstChild; c != nil; c = c.NextSibling {
+			p.addSourceMapping(c.Loc[0])
 			if c.Type == TextNode {
-				p.addSourceMapping(c.Loc[0])
 				p.print(c.Data)
 			} else {
-				p.addSourceMapping(c.Loc[0])
-				if (c.PrevSibling == nil || (c.PrevSibling != nil && c.PrevSibling.Type == TextNode)) && c.NextSibling != nil {
+				if c.PrevSibling == nil || (c.PrevSibling != nil && c.PrevSibling.Type == TextNode) {
```
Comment (drwpow) [inline internal/printer/print-to-js.go:310]: This was the expression I removed: `c.NextSibling != nil`. Not sure if that was a logical bug, or if I just introduced a regression. But the test added seemed to pass now.

File: internal/printer/printer_test.go [hunk 2]
```diff
@@ -96,6 +96,16 @@ import VueComponent from '../components/Vue.vue';
...
 			},
 		},
+		{
+			name:   "ternary",
+			source: `<body>{false ? <div>#f</div> : <div>#t</div>}</body>`,
+			want: want{
+				imports:     "",
+				frontmatter: []string{},
+				styles:      []string{},
+				code:        "<html><head></head><body>${false ? $$render`<div>#f</div>` : $$render`<div>#t</div>`}</body></html>",
```
Comment (drwpow) [inline internal/printer/printer_test.go:106]: This was what it was rendering before:

```
${false ? $$render`<div>#f</div>` : <div>#t</div>`}
```

Notice the `$$render` and backtick are missing from the right-hand side

File: internal/printer/print-to-js.go (hunk 1 above)
Comment (natemoo-re) [reply internal/printer/print-to-js.go:310]: I feel like I ran into a condition where that was needed, but I can't remember what it was.

**Review by natemoo-re — APPROVED** [review]
Good to merge!

Would love to follow-up with more expression tests... Maybe we can catch some other bugs lurking in this area. (I'm sure there are some)

**Comment by drwpow** [comment]
> Maybe we can catch some other bugs lurking in this area. (I'm sure there are some)

Oh don’t worry! There are plenty more where this came from 😛 (just uncomment some tests in the `next` branch)

**Pull request merged**